azkaban-common/src/main/java/azkaban/imagemgmt/rampup/ImageRampupManagerImpl.java

```diff
@@ -107,16 +159,69 @@ public ImageRampupManagerImpl(final ImageRampupDao imageRampupDao,
    * Follow the rampup procedure to elect a new version from the image_versions table for the ailed
    * image type.
    *
-   * @param imageTypes       - set of specified image types
-   * @param imageTypeRampups - contains rampup list for an image type
-   * @return Map<String, String>
+   * @param imageTypes          - set of specified image types
+   * @param imageTypeRampups    - contains rampup list for an image type
+   * @param remainingImageTypes - This set is used to keep track of the image types for which
+   *                              version metadata is not available.
+   * @return Map<String, VersionMetadata>
+   */
+  private Map<String, ImageVersionMetadata> processAndGetVersionForImageTypes(
```

Comment:
Is there a 'randomness' in the result of this method across successive invocations?
Also can there be some disconnect b/w the values returned by this method as part of a user facing API call (json result) and what values are actually picked for the rampup?

Reply:
Thanks for pointing this out. Yes random version will be selected based on rampup logic. If there is no rampup information latest available active version will be selected. As these information are shown on user facing status API, I have included a new property called "message" to order to indicate how the displayed version is selected. This should clear any confusion around this .